backend/capellacollab/sessions/routes.py

```diff
@@ -189,13 +204,58 @@ def request_persistent_session(
             },
         )
 
-    docker_image = get_image_for_tool_version(db, body.version)
+    tool = get_existing_tool(body.tool_id, db)
+    version = get_exisiting_tool_version(tool.id, body.version_id, db)
+
+    docker_image = get_image_for_tool_version(db, version.id)
+
+    admin_stmt = (
+        select(DatabaseT4CRepository)
+        .join(DatabaseT4CRepository.models)
+        .join(DatabaseT4CModel.model)
+        .where(DatabaseCapellaModel.tool == tool)
+        .where(DatabaseCapellaModel.version == version)
+    )
+
+    stmt = (
+        admin_stmt.join(DatabaseCapellaModel.project)
+        .join(DatabaseProject.users)
+        .where(
+            ProjectUserAssociation.permission == ProjectUserPermission.WRITE
+        )
+        .where(ProjectUserAssociation.user == user)
+    )
+
+    t4c_repositories: t.Optional[list[DatabaseT4CRepository]] = (
+        (
+            db.execute(admin_stmt if user.role == Role.ADMIN else stmt)
+            .scalars()
+            .all()
+        )
+        if tool.name == "Capella"
+        else None
+    )
```

Comment:
Do we want to do database queries directly in the router? This should probably move to one of the crud/model modules.

Reply:
I left the query there because I didn’t know to which crud it should belong, because it’s across many tables, and requires a lot of external objects. But Since the object returned is a `DatabaseT4CRepository`, it can actually be moved to _projects/capellamodels/modelsources/t4c/crud.py_. However, I still feel that it doesn’t really belong to a crud, because it’s a very specific query, that doesn’t intend to be reused.